core/src/main/scala/trees/Trees.scala

```diff
@@ -865,6 +881,160 @@ object Tree {
       }
     rec(t)
   }
+
+  def subTrees(t: Tree): List[Tree] = {
+    t match {
+      case Var(id) => List()
+      case NatLiteral(n) => List()
+      case Succ(t) => List(t)
+      case UnitLiteral => List()
+      case BooleanLiteral(b) => List()
+      case Bind(id, body) => List(body)
+      case IfThenElse(cond, t1, t2) => List(cond, t1, t2)
+      case Lambda(tp, bind) => tp.toList ++ List(bind)
+      case DefFunction(args, optRet, optMeasure, body, rest) => ???// List(body, rest) ++ optMeasure.toList ++ optRet.toList 
+      case ErasableLambda(ty, bind) => ???
+      case App(t1, t2) => List(t2,t1)//List(t1, t2)
+      case Pair(t1, t2) => List(t1, t2)
+      case Size(t) => List(t)
+      case First(t) => List(t)
+      case Second(t) => List(t)
+      case Fix(tp, bind) => tp.toList ++ List(bind)
+      case NatMatch(t, t1, t2) => List(t, t1, t2)
+      case EitherMatch(t, t1, t2) => List(t, t1, t2)
+      case LeftTree(t) => List(t)
+      case RightTree(t) => List(t)
+      case LetIn(tp, v, body) => tp.toList ++ List(v, body) 
+      case MacroTypeDecl(tp, body) => List(tp, body)
+      case MacroTypeInst(v, args) => ???
+      case Error(s, t) => t.toList
+      case Primitive(op, args) => args
+      case ErasableApp(t1, t2) => List(t1, t2)
+      case Refl(t1, t2) => List(t1, t2)
+      case Fold(tp, t) => List(tp, t)
+      case Unfold(t, bind) => List(t, bind)
+      case UnfoldPositive(t, bind) => List(t, bind)
+      case Abs(t) => List(t)
+      case TypeApp(t1, t2) => List(t1, t2)
+      case BottomType => List()
+      case TopType => List()
+      case UnitType => List()
+      case BoolType => List()
+      case NatType => List()
+      case SigmaType(t1, t2) => List(t1, t2)
+      case SumType(t1, t2) => List(t1, t2)
+      case PiType(t1, t2) => List(t1, t2)
+      case IntersectionType(t1, t2) => List(t1, t2)
+      case ExistsType(t1, t2) => List(t1, t2)
+      case RefinementType(t1, t2) => List(t1, t2)
+      case RefinementByType(t1, t2) => List(t1, t2)
+      case RecType(n, bind) => List(n, bind)
+      case PolyForallType(t) => List(t)
+      case UnionType(t1, t2) => List(t1, t2)
+      case EqualityType(t1, t2) => List(t1, t2)
+      case Because(t1, t2) => List(t1, t2)
+      case Node(name, children) => children.toList
+    }
+  }
+
+  def newSubTrees(t: Tree, args: List[Tree])(implicit rc: RunContext): Tree = {
```

Comment:
This name is not ideal, any ideas on better ones ?

Reply:
What does this do? (Can you add a comment?)